cmd/yggctl/generate_test.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/redhatinsights/yggdrasil"
+)
+
+type Input struct {
+	messageType string
+	responseTo  string
+	directive   string
+	content     []byte
+	metadata    map[string]string
+	version     int
+}
+
+func TestGenerateDataMessage(t *testing.T) {
+	tests := []struct {
```

Comment:
Have you considered using [Ginkgo](https://onsi.github.io/ginkgo/) and especially [table driven tests](https://pkg.go.dev/github.com/onsi/ginkgo/extensions/table?utm_source=godoc)? 
To me tests written with plain Go are pain to read.

Reply:
I have. I don't like taking on additional package dependencies without good reason, and adding a large module like Ginkgo as a dependency purely for testing isn't worth it. I also don't like the "language" of behavior-driven tests as much as table-driven tests; to me, they mask the actual code behind syntax sugar and make it more obtuse when trying to step-debug the tests. I find behavior-driven tests a pain to read. How about that!

Reply:
:D